src/execution_plan/optimizations/select_entry_point.c

```diff
@@ -7,6 +7,8 @@
 #include "./select_entry_point.h"
 
 void selectEntryPoint(AlgebraicExpression *ae, const FT_FilterNode *tree) {
+    if (ae->src_node == ae->dest_node) return;
```

Comment:
In `_AlgebraicExpression_IsolateVariableLenExps`, if the source or destination label matrix is labeled, it's put into a separate expression. It's connected to another AlgebraicExpression if possible, but is otherwise left dangling. In this case, the AlgebraicExpression is a single operand with the same source and destination:
https://github.com/RedisGraph/RedisGraph/blob/master/src/arithmetic/algebraic_expression.c#L126
https://github.com/RedisGraph/RedisGraph/blob/master/src/arithmetic/algebraic_expression.c#L146

In this file, this is just a quick return. The changes in `traverse_order.c` are more important, as they ensure that a pattern like:
`MATCH (a)-[*]->(b:is_labeled)` properly creates a variable-length traversal and a label scan (which is a long-standing issue).

Reply:
As long as we can't reuse entities this check if OK, consider (a)-[]->(b)-[]->(a)